tests/unit/net/fetch.rs

```diff
@@ -556,6 +555,57 @@ fn test_fetch_with_hsts() {
                "https");
 }
 
+#[test]
+fn test_fetch_with_sri_sucess() {
+    static MESSAGE: &'static [u8] = b"alert('Hello, world.');";
+    let handler = move |_: HyperRequest, response: HyperResponse| {
+        response.send(MESSAGE).unwrap();
+    };
+    let (mut server, server_url) = make_server(handler);
+    let do_fetch = |url: ServoUrl| {
+        let origin = Origin::Origin(url.origin());
+        let mut request = Request::new(url, Some(origin), false, None);
+        *request.referrer.borrow_mut() = Referrer::NoReferrer;
+        *request.integrity_metadata.borrow_mut()
+             = "sha384-H8BRh8j48O9oYatfu5AZzq6A9RINhZO5H16dQZngK7T62em8MUt1FLm52t+eX6xO".to_owned();
+        // Set the flag.
+        request.local_urls_only = false;
+        fetch(request, None)
+    };
+
+    let response = do_fetch(server_url);
+
+    let _ = server.close();
+    assert_eq!(response_is_done(&response), true);
+}
+
+#[test]
+fn test_fetch_with_sri_network_error() {
+    static MESSAGE: &'static [u8] = b"alert('Hello, Network Error');";
+    let handler = move |_: HyperRequest, response: HyperResponse| {
+        response.send(MESSAGE).unwrap();
+    };
+    let (mut server, server_url) = make_server(handler);
+
+    let do_fetch = |url: ServoUrl| {
+        let origin = Origin::Origin(url.origin());
+        let mut request = Request::new(url, Some(origin), false, None);
+        *request.referrer.borrow_mut() = Referrer::NoReferrer;
+        *request.integrity_metadata.borrow_mut()
+             = "sha384-H8BRh8j48O9oYatfu5AZzq6A9RINhZO5H16dQZngK7T62em8MUt1FLm52t+eX6xO".to_owned();
+        // Set the flag.
+        request.local_urls_only = false;
+
+        fetch(request, None)
+    };
+
+    let response = do_fetch(server_url);
+
+    let _ = server.close();
+    assert!(response.is_network_error());
+    assert!(response.internal_response.unwrap().is_network_error());
```

Comment:
This assertion won't pass after previous changes I requested. We should be able to remove it.